pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -211,7 +211,7 @@ func startMemoryThresholdNotifier(thresholds []evictionapi.Threshold, observatio
 
 // synchronize is the main control loop that enforces eviction thresholds.
 // Returns the pod that was killed, or nil if no pod was killed.
-func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc ActivePodsFunc, nodeProvider NodeProvider) []*v1.Pod {
+func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc ActivePodsFunc, containerManager cm.ContainerManager) []*v1.Pod {
```

Comment:
Why are you making this change instead of extending the NodeProvider interface to include the additional methods you need and pass a container manager object that implements NodeProvider interface?